lib/include/openturns/OTtestcode.hxx

```diff
@@ -400,6 +403,177 @@ void assert_equal(const T & a, const T & b, const String errMsg = "")
   }
 }
 
+class IshigamiUseCase
```

Comment:
I dont know if we want to duplicate the use-cases into c++, I think its fine to have high-level tests only in Python only

Reply:
Even if that means duplicating the Ishigami unit test over dozens of C++ unit tests?

Reply:
yes yes ok